src/pip/_internal/utils/unpacking.py

```diff
@@ -0,0 +1,286 @@
+# mypy: strict-optional=False
+
+from __future__ import absolute_import
+
+import logging
+import mimetypes
+import os
+import shutil
+import stat
+import tarfile
+import zipfile
+
+from pip._internal.exceptions import InstallationError
+from pip._internal.utils.misc import (
+    current_umask,
+    ensure_dir,
+    file_contents,
+    has_leading_dir,
```

Comment:
`is_svn_page()`, `has_leading_dir()` and `split_leading_dir()` should also be moved to the new module, as the only place they're used is in the unpack functions.